app/scripts/controllers/network/createInfuraClient.test.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * @jest-environment node
+ */
+
+import { withInfuraClient } from './provider-api-tests/helpers';
+import {
+  testsForRpcMethodNotHandledByMiddleware,
+  testsForRpcMethodAssumingNoBlockParam,
+  testsForRpcMethodsThatCheckForBlockHashInResponse,
+  testsForRpcMethodSupportingBlockParam,
+} from './provider-api-tests/shared-tests';
+
+describe('createInfuraClient', () => {
+  // Infura documentation: <https://docs.infura.io/infura/networks/ethereum/json-rpc-methods>
+  // Ethereum JSON-RPC spec: <https://ethereum.github.io/execution-apis/api-documentation/>
+
+  describe('RPC methods supported by Infura and listed in the JSON-RPC spec', () => {
+    describe('eth_accounts', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_accounts', {
+        numberOfParameters: 0,
+      });
+    });
+
+    describe('eth_blockNumber', () => {
+      testsForRpcMethodAssumingNoBlockParam('eth_blockNumber');
+    });
+
+    describe('eth_call', () => {
+      testsForRpcMethodSupportingBlockParam('eth_call', {
+        blockParamIndex: 1,
+      });
+    });
+
+    describe('eth_chainId', () => {
+      it('does not hit Infura, instead returning the chain id that maps to the Infura network, as a hex string', async () => {
+        const chainId = await withInfuraClient(
+          { network: 'ropsten' },
+          ({ makeRpcCall }) => {
+            return makeRpcCall({
+              method: 'eth_chainId',
+            });
+          },
+        );
+
+        expect(chainId).toStrictEqual('0x3');
+      });
+    });
+
+    describe('eth_coinbase', () => {
+      testsForRpcMethodNotHandledByMiddleware('eth_coinbase', {
+        numberOfParameters: 0,
+      });
+    });
+
+    describe('eth_estimateGas', () => {
+      testsForRpcMethodAssumingNoBlockParam('eth_estimateGas', {
+        numberOfParameters: 0,
```

Comment:
This function doesn't take any options. Just the RPC method

Reply:
Whoops! Fixed in [332a958](https://github.com/MetaMask/metamask-extension/pull/15556/commits/332a95825eb636351d7e080c067aed317e918b38).